app/src/main/java/com/teamagam/gimelgimel/app/common/ExponentialBackoffStrategy.java

```diff
@@ -0,0 +1,49 @@
+package com.teamagam.gimelgimel.app.common;
+
+/**
+ * Exponential backoff strategy implementation.
+ * Time intervals grows exponentially between increases.
+ */
+public class ExponentialBackoffStrategy implements BackoffStrategy {
+
+    private final int mMultiplier;
+    private final int mMaxBackoffMillis;
+    private final int mBaseIntervalMillis;
+
+    private long mCurrentIntervalMillis;
```

Comment:
why do you need both mCurrentIntervalMillis and mCurrentBackoffMillis? just multply each step with mMultiplier. you dont need to add mCurrentIntervalMillis.